iwyu_output.cc

```diff
@@ -456,9 +456,16 @@ string MungedForwardDeclareLineForTemplates(const TemplateDecl* decl) {
   // The template name is now the last word on the line.  Replace it
   // by its fully-qualified form.  Apparently rfind's endpos
   // argument is inclusive, so substract one to get past the end-space.
-  const string::size_type name = line.rfind(' ', endpos - 1);
+  const char kFinalSpecifier[] = " final ";
+  string::size_type final_pos = line.find(kFinalSpecifier);
+  if (final_pos != string::npos) {
+    line.replace(final_pos, sizeof(kFinalSpecifier), " ");
+  }
+  string::size_type name = line.rfind(" ", endpos - 1);
```

Comment:
I was worried that `endpos` was no longer correct after the edits for `final`, but it seems it's never been relevant.

I tried removing it locally on master, and all tests still pass. I think I'll push a refactor commit to get rid of that, and maybe you can rebase?

Reply:
If you leave the `' '` space literal as a char instead of a string literal, the diff will be smaller, and it will probably generate better code.